### Objective- To define a function and find the model accuracy.


In [2]:
#importing the required library
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
%matplotlib inline


C:\Users\DELL\anaconda3\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)


In [6]:
#importing the dataset
dataset=pd.read_csv("Titanic_train.csv")
dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [8]:
#droping the unnecessary column
dataset=dataset.drop(['PassengerId','Name','Ticket','Cabin'], axis=1)
dataset.head()

Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0         0       3    male  22.0      1      0   7.2500        S
1         1       1  female  38.0      1      0  71.2833        C
2         1       3  female  26.0      0      0   7.9250        S
3         1       1  female  35.0      1      0  53.1000        S
4         0       3    male  35.0      0      0   8.0500        S

In [9]:
#Converting the categoriacl to numeric
from sklearn.preprocessing import LabelEncoder
le= LabelEncoder()
dataset['Sex']=le.fit_transform(dataset['Sex'])
dataset['Embarked']=le.fit_transform(dataset['Embarked'])
dataset.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

### Defining Function to find the accuracy score and confusion matrix


In [14]:
def fun_model(a,b):
    from sklearn.model_selection import train_test_split
    from sklearn.metrics import accuracy_score, confusion_matrix
    from sklearn import svm
    c=svm.SVC(gamma=0.01,C=100)
    x_train,x_test,y_train,y_test=train_test_split(a,b,test_size=0.3,random_state=0)
    c.fit(x_train,y_train)
    y_pred=c.predict(x_test)
    acc=accuracy_score(y_test,y_pred)*100
    con=confusion_matrix(y_test,y_pred)
    return acc,con

### Model 1: Taking PCLass as Dependent and rest as independent variable

In [11]:
y=dataset.iloc[:,1:2]
y.head()

Pclass
0       3
1       1
2       3
3       1
4       3

In [12]:
x=dataset.drop(['Pclass'],axis=1)
x.head()

Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0         0    1  22.0      1      0   7.2500         2
1         1    0  38.0      1      0  71.2833         0
2         1    0  26.0      0      0   7.9250         2
3         1    0  35.0      1      0  53.1000         2
4         0    1  35.0      0      0   8.0500         2

#### Evaulating the model

In [15]:
AccuracyScore,ConfusionMatrix=fun_model(x,y)
print("The accuracy score is:",AccuracyScore)
print("The counfusion matrix is:")
print(ConfusionMatrix)

The accuracy score is: 90.2621722846442
The counfusion matrix is:
[[ 64   4   2]
 [  4  36   9]
 [  1   6 141]]


C:\Users\DELL\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Inference-**
 1. The Accuracy score is **90.26%**
 2. **241** out of **267** are correctly Classified



### Model 2: Sex as the dependent variable and others as independent variable

In [16]:
y1=dataset.iloc[:,2:3]
y1.head()

Sex
0    1
1    0
2    0
3    0
4    1

In [17]:
x1=dataset.drop(['Sex'],axis=1)
x1.head()

Survived  Pclass   Age  SibSp  Parch     Fare  Embarked
0         0       3  22.0      1      0   7.2500         2
1         1       1  38.0      1      0  71.2833         0
2         1       3  26.0      0      0   7.9250         2
3         1       1  35.0      1      0  53.1000         2
4         0       3  35.0      0      0   8.0500         2

#### Evaulting the model

In [18]:
AccuracyScore,ConfusionMatrix=fun_model(x1,y1)
print("Accuracy Score is", AccuracyScore)
print("Confusion Matrix is")
print(ConfusionMatrix)

Accuracy Score is 72.65917602996255
Confusion Matrix is
[[ 66  32]
 [ 41 128]]


C:\Users\DELL\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Inference-**
 1. The Accuracy score is **72.65%**
 2. **194** out of **267** are correctly Classified



### Model 3: Sibsp VS all other

In [21]:
y2=dataset.iloc[:,4:5]
y2.head()

SibSp
0      1
1      1
2      0
3      1
4      0

In [22]:
x2=dataset.drop(["SibSp"],axis=1)
x2.head()

Survived  Pclass  Sex   Age  Parch     Fare  Embarked
0         0       3    1  22.0      0   7.2500         2
1         1       1    0  38.0      0  71.2833         0
2         1       3    0  26.0      0   7.9250         2
3         1       1    0  35.0      0  53.1000         2
4         0       3    1  35.0      0   8.0500         2

#### Evaulation the model

In [23]:
AccuracyScore,ConfusionMatrix=fun_model(x2,y2)
print("Accuracy Score is",AccuracyScore)
print("Confusion Matrix is")
print(ConfusionMatrix)

Accuracy Score is 74.90636704119851
Confusion Matrix is
[[150  24   7   0   1   0   0]
 [ 21  39   5   0   0   0   0]
 [  4   1   3   0   0   0   0]
 [  0   2   1   3   1   0   0]
 [  0   0   0   0   2   0   0]
 [  0   0   0   0   0   1   0]
 [  0   0   0   0   0   0   2]]


C:\Users\DELL\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Inference-**
 1. The Accuracy score is **74.90%**
 2. **200** out of **267** are correctly Classified



### Model 4: Parch Vs all others

In [24]:
y3=dataset.iloc[:,5:6]
y3.head()

Parch
0      0
1      0
2      0
3      0
4      0

In [25]:
x3=dataset.drop(["Parch"],axis=1)
x3.head()

Survived  Pclass  Sex   Age  SibSp     Fare  Embarked
0         0       3    1  22.0      1   7.2500         2
1         1       1    0  38.0      1  71.2833         0
2         1       3    0  26.0      0   7.9250         2
3         1       1    0  35.0      1  53.1000         2
4         0       3    1  35.0      0   8.0500         2

#### Evaulating the model

In [26]:
AccuracyScore,ConfusionMatrix=fun_model(x3,y3)
print("Accuracy Score is", AccuracyScore)
print("Confusion Matrix")
print(ConfusionMatrix)

Accuracy Score is 79.7752808988764
Confusion Matrix
[[187   6   5   0   0   0]
 [ 21  15   3   0   0   1]
 [ 10   6  11   0   0   0]
 [  0   0   1   0   0   0]
 [  1   0   0   0   0   0]
 [  0   0   0   0   0   0]]


C:\Users\DELL\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Inference-**
 1. The Accuracy score is **79.77%**
 2. **213**  out of 267 are correctly Classified



### Model 5: Embarked Vs all

In [27]:
x4=dataset.drop(["Embarked"],axis=1)
x4.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare
0         0       3    1  22.0      1      0   7.2500
1         1       1    0  38.0      1      0  71.2833
2         1       3    0  26.0      0      0   7.9250
3         1       1    0  35.0      1      0  53.1000
4         0       3    1  35.0      0      0   8.0500

In [30]:
y4=dataset.iloc[:,-1:]
y4.head()

Embarked
0         2
1         0
2         2
3         2
4         2

#### Evaulting the model.


In [32]:
AccuracyScore,ConfusionMatrix=fun_model(x4,y4)
print("Accuracy Score is", AccuracyScore)
print("Confusion Matrix")
print(ConfusionMatrix)

Accuracy Score is 76.77902621722846
Confusion Matrix
[[ 23   2  24]
 [  0   8  14]
 [ 18   4 174]]


C:\Users\DELL\anaconda3\lib\site-packages\sklearn\utils\validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


**Inference-**
 1. The Accuracy score is **76%**
 2. **205** out of **267** are correctly Classified